### PR TITLE
Combine ESLint and npm audit workflows into a reusable template

### PR DESCRIPTION
**Pull request by gupta-soham**
## 📋 Description

This pull request combines our separate ESLint and npm audit workflows into a single, reusable workflow template. The new workflow is designed to be used across multiple UI repositories within our PSMRI organization, improving consistency and maintainability of our CI/CD processes.

---

## ✅ Type of Change

- [ ] 🐞 Bug fix (non-breaking change which resolves an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🔥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🛠 Refactor (change that is neither a fix nor a new feature)
- [x] ⚙️ Config change (configuration file or build script updates)
- [x] 📚 Documentation (updates to docs or readme)
- [ ] 🧪 Tests (adding new or updating existing tests)
- [ ] 🎨 UI/UX (changes that affect the user interface)
- [x] 🚀 Performance (improves performance)
- [ ] 🧹 Chore (miscellaneous changes that don't modify src or test files)

---

### Key Updates:
1. Created a new `lint-audit.yml` workflow template in the `.github/workflow-templates/` directory.
2. Combined ESLint and npm audit jobs into a single workflow.
3. Implemented caching for npm dependencies to improve workflow speed.
4. Added selective linting for changed files only.
5. Included artifact upload for npm audit reports.
6. Updated the workflow to accept `workflow_call` events for reusability.
7. Added comprehensive comments for better maintainability.

### Implementation Details:
- The new workflow template is located at `.github/workflow-templates/lint-audit.yml`.
- Updated `lint-audit.properties.json` to reflect the combined workflow.

### Next Steps:
1. Merge this PR into the `.github` repository.
2. Update all UI repositories to use this new workflow template.
3. Monitor initial runs across repositories and adjust as needed.

Please review and let me know if any further changes are needed.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new "Lint and Audit" workflow template for automated linting and security audits of JavaScript and TypeScript projects.
	- Added a GitHub Actions workflow that runs linting and auditing jobs on code changes, enhancing CI/CD processes.

- **Bug Fixes**
	- Removed outdated workflow templates to streamline the process.

- **Documentation**
	- Updated descriptions and categorizations for better clarity on the purpose of the new workflows.

### CODE REVIEW

**Comment by gupta-soham** [comment]
Reference: https://github.com/PSMRI/AMRIT/issues/3#issuecomment-2409058878

File: workflow-templates/lint_audit.yml
Comment (drtechie) [inline workflow-templates/lint_audit.yml:1]: Ubuntu will be initialized twice, which may result in additional execution time. 
@gupta-soham can you combine both lint and audit to 'steps' under the same job?

This PR right now doesn't give any additional benefits compared to separate files.

**Review by drtechie — CHANGES_REQUESTED** [review]
Combine lint and audit as steps

**Comment by gupta-soham** [comment]
I have combined the lint and audit in the same step. @drtechie let me know if any other changes are required.

**Review by drtechie — APPROVED** [review]
Can remove --force from npm install

**Pull request merged**